bindings/wasm/src/macros.rs

```diff
@@ -7,3 +7,16 @@ macro_rules! log {
     web_sys::console::log_1(&format!($($tt)*).into());
   }
 }
+
+#[macro_export]
+macro_rules! impl_wasm_clone {
+  ($wasm_class:ident, $js_class:ident) => {
+    #[wasm_bindgen(js_class = $js_class)]
+    impl $wasm_class {
+      #[wasm_bindgen(js_name = clone)]
+      pub fn deep_clone(&self) -> $wasm_class {
+        return $wasm_class(self.0.clone());
+      }
+    }
```

Comment:
Nit: we should document that this is a deep clone.

```suggestion
    impl $wasm_class {
      /// Deep clones the object.
      #[wasm_bindgen(js_name = clone)]
      pub fn deep_clone(&self) -> $wasm_class {
        return $wasm_class(self.0.clone());
      }
    }
```